lms/djangoapps/verify_student/tests/test_ssencrypt.py

```diff
@@ -72,13 +52,38 @@ def test_rsa():
 l8N6+LEIVTMAytPk+/bImHvGHKZkCz5rEMSuYJWOmqKI92rUtI6fz5DUb3XSbrwT
 3W+sdGFUK3GH1NAX71VxbAlFVLUetcMwai1+wXmGkRw6A7YezVFnhw==
 -----END RSA PRIVATE KEY-----"""
-    aes_key_str = "32fe72aaf2abb44de9e161131b5435c8d37cbdb6f5df242ae860b283115f2dae"
 
-    aes_key = aes_key_str.decode('hex')
 
-    encrypted_aes_key = rsa_encrypt(aes_key, pub_key_str)
-    assert_equals(aes_key, rsa_decrypt(encrypted_aes_key, priv_key_str))
+def test_aes():
+    def assert_roundtrip(text):
+        assert aes_decrypt(aes_encrypt(text, AES_KEY_BYTES), AES_KEY_BYTES) == text
+        assert decode_and_decrypt(encrypt_and_encode(text, AES_KEY_BYTES), AES_KEY_BYTES) == text
+
+    assert_roundtrip(b"Hello World!")
+    assert_roundtrip(b"1234567890123456")  # AES block size, padding corner case
+    # Longer string
+    assert_roundtrip(b"12345678901234561234567890123456123456789012345601")
+    assert_roundtrip(b"")
+    assert_roundtrip(b"\xe9\xe1a\x13\x1bT5\xc8")  # Random, non-ASCII text
+
+
+def test_rsa():
+    _assert_rsa(AES_KEY_BYTES, PUB_KEY_BYTES, PRIV_KEY_BYTES)
+
+
+def test_rsa_unicode_data():
+    data = u'12345678901234567890123456789012'
+    _assert_rsa(data, PUB_KEY_BYTES, PRIV_KEY_BYTES)
+
+
+def test_rsa_unicode_keys():
+    _assert_rsa(AES_KEY_BYTES, PUB_KEY_BYTES.decode('utf-8'), PRIV_KEY_BYTES.decode('utf-8'))
+
+
+def _assert_rsa(data, public_key, private_key):
```

Comment:
Nit: Could have just DDTed this method instead.

Reply:
That hadn't occurred to me, although I think the resulting test case names would be huge after being mangled to include the parameters.